### PR TITLE
Added exceptions thrown from subroutines into the xml documentation comments of ValidateToken()

### PR DESCRIPTION
**Pull request by mafurman**
Addresses #780.

### CODE REVIEW

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 1]
```diff
@@ -740,8 +759,15 @@ public virtual ClaimsPrincipal ValidateToken(string token, TokenValidationParame
...
             else
             {
-                validatedToken = ValidateSignature(token, validationParameters);
-                return ValidateTokenPayload(validatedToken as JwtSecurityToken, validationParameters);
+                try
+                {
+                    validatedToken = ValidateSignature(token, validationParameters);
+                    return ValidateTokenPayload(validatedToken as JwtSecurityToken, validationParameters);
+                }
+                catch (FormatException e)
```
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: move this try catch inside ValidateSignature.

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 2]
```diff
@@ -711,8 +711,27 @@ public override SecurityToken ReadToken(XmlReader reader, TokenValidationParamet
...
+        /// <exception cref="SecurityTokenEncryptionKeyNotFoundException">if 'jwtToken.Header.kid' is not null AND decryption fails.</exception>
+        /// <exception cref="SecurityTokenDecryptionFailedException">if the JWE was not able to be decrypted.</exception>
+        /// <exception cref="SecurityTokenInvalidSignatureException">if the signature is not properly formatted.</exception>
+        /// <exception cref="SecurityTokenNoExpirationException">If 'expires.HasValue' is false and <see cref="TokenValidationParameters.RequireExpirationTime"/> is true.</exception>
+        /// <exception cref="SecurityTokenInvalidLifetimeException">If 'notBefore' is &gt; 'expires'.</exception>
+        /// <exception cref="SecurityTokenNotYetValidException">If 'notBefore' is &gt; DateTime.UtcNow.</exception>
+        /// <exception cref="SecurityTokenExpiredException">If 'expires' is &lt; DateTime.UtcNow.</exception>
+        /// <exception cref="SecurityTokenInvalidAudienceException">If <see cref="TokenValidationParameters.ValidAudience"/> is null or whitespace and <see cref="TokenValidationParameters.ValidAudiences"/> is null.</exception>
```
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: put exceptions in order

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs (hunk 1 above)
Comment (mafurman) [reply src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: Done!

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs (hunk 2 above)
Comment (mafurman) [reply src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: Done!

**Comment by brentschmaltz** [comment]
:shipit:

@vanillajonathan  if you see something let us know.

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 3]
```diff
@@ -707,12 +707,31 @@ public override SecurityToken ReadToken(XmlReader reader, TokenValidationParamet
...
+        /// <exception cref="SecurityTokenNoExpirationException">If <see cref="TokenValidationParameters.TokenReplayCache"/> is not null and expirationTime.HasValue is false. When a TokenReplayCache is set, tokens require an expiration time.</exception>
+        /// <exception cref="SecurityTokenNotYetValidException">If 'notBefore' is &gt; DateTime.UtcNow.</exception>
+        /// <exception cref="SecurityTokenReplayAddFailedException">If the 'securityToken' could not be added to the <see cref="TokenValidationParameters.TokenReplayCache"/>.</exception>
+        /// <exception cref="SecurityTokenReplayDetectedException">If the 'securityToken' is found in the cache.</exception>
+        /// <returns> A <see cref="ClaimsPrincipal"/> from the jwt. Does not include the header claims.</returns>
+        /// <remarks> 
+        /// Many of the exceptions listed above are not thrown directly from this method. See <class cref="Validators"/> to find most of the call graph.
+        /// <see cref = "ReadToken(string)" /> for additional exceptions.
```
Comment (vanillajonathan) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: There are spaces around `cref`.

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 4]
```diff
@@ -707,12 +707,31 @@ public override SecurityToken ReadToken(XmlReader reader, TokenValidationParamet
...
         /// <param name="validatedToken">The <see cref="JwtSecurityToken"/> that was validated.</param>
-        /// <exception cref="ArgumentNullException">'token' is null or whitespace.</exception>
-        /// <exception cref="ArgumentNullException">'validationParameters' is null.</exception>
         /// <exception cref="ArgumentException">token.Length > MamimumTokenSizeInBytes.</exception>
         /// <exception cref="ArgumentException">'token' does not have 3 or 5 parts.</exception>
-        /// <returns>A <see cref="ClaimsPrincipal"/> from the jwt. Does not include the header claims.</returns>
-        /// <remarks><see cref="ReadToken(string)"/> for additional exceptions.</remarks>
+        /// <exception cref="ArgumentException">'token.Length' > MaximumTokenSizeInBytes.</exception>
+        /// <exception cref="ArgumentException"><see cref="CanReadToken(string)"/></exception>
+        /// <exception cref="ArgumentNullException">'token' is null or whitespace.</exception>
```
Comment (vanillajonathan) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: 'token' does not uses `<paramref>` markup.

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 5]
```diff
@@ -707,12 +707,31 @@ public override SecurityToken ReadToken(XmlReader reader, TokenValidationParamet
...
+        /// <exception cref="SecurityTokenInvalidSignatureException">if the signature is not properly formatted.</exception>
+        /// <exception cref="SecurityTokenNoExpirationException">If 'expires.HasValue' is false and <see cref="TokenValidationParameters.RequireExpirationTime"/> is true.</exception>
+        /// <exception cref="SecurityTokenNoExpirationException">If <see cref="TokenValidationParameters.TokenReplayCache"/> is not null and expirationTime.HasValue is false. When a TokenReplayCache is set, tokens require an expiration time.</exception>
+        /// <exception cref="SecurityTokenNotYetValidException">If 'notBefore' is &gt; DateTime.UtcNow.</exception>
+        /// <exception cref="SecurityTokenReplayAddFailedException">If the 'securityToken' could not be added to the <see cref="TokenValidationParameters.TokenReplayCache"/>.</exception>
+        /// <exception cref="SecurityTokenReplayDetectedException">If the 'securityToken' is found in the cache.</exception>
+        /// <returns> A <see cref="ClaimsPrincipal"/> from the jwt. Does not include the header claims.</returns>
```
Comment (vanillajonathan) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: JWT is an acronym and hence should properly capitalized.

**Comment by brentschmaltz** [comment]
@vanillajonathan touched up comments.

**Pull request merged**